src/main/sensors/barometer.c

```diff
@@ -449,15 +447,18 @@ uint32_t baroUpdate(timeUs_t currentTimeUs)
 
             // update baro data
             baro.dev.calculate(&baro.pressure, &baro.temperature);
-            baro.altitude = pressureToAltitude(baro.pressure);
-
-            if (baroIsCalibrated()) {
-                // zero baro altitude
-                baro.altitude -= baroGroundAltitude;
-            } else {
-                // establish stable baroGroundAltitude value to zero baro altitude with
-                performBaroCalibrationCycle(baro.altitude);
-                baro.altitude = 0.0f;
+            // If baro.pressure is invalid then skip altitude counting / call of calibration cycle
+            if (baro.pressure > 0) {
+                baro.altitude = pressureToAltitude(baro.pressure);
+
+                if (baroIsCalibrated()) {
+                    // zero baro altitude
+                    baro.altitude -= baroGroundAltitude;
+                } else {
+                    // establish stable baroGroundAltitude value to zero baro altitude with
+                    performBaroCalibrationCycle(baro.altitude);
+                    baro.altitude = 0.0f;
+                }
             }
```

Comment:
```suggestion
            // If baro.pressure is invalid then skip altitude counting / call of calibration cycle
            if (baro.pressure > 0) {
                const float altitude = pressureToAltitude(baro.pressure);

                if (baroIsCalibrated()) {
                    // zero baro altitude
                    baro.altitude = altitude - baroGroundAltitude;
                } else {
                    // establish stable baroGroundAltitude value to zero baro altitude with
                    performBaroCalibrationCycle(altitude);
                    baro.altitude = 0.0f;
                }
            } else {
                 if (!baroIsCalibrated()) { // return 0 during calibration, reuse last value otherwise
                      baro.altitude = 0.0f;
                 }
            }
```

It is slightly better to write global variables only in one place. ```

Reply:
I agree, I missed that in case calibration started and baro begins providing wrong values same time then non-zero altitude will be fixed for a while.

Reply:
In case baro providing wrong data we will continue gathering calibration samples with last correct altitude and after that with 0.0f altitude. Anyway baroGroundAltitude will be wrong.
Also if baro is calibrated but pressure is wrong - we will start calibration cycle - not needed.

Reply:
Thanks @ledvinap and @Neolight79 ❤️